src/camp/generators/lib/campch.py

```diff
@@ -150,6 +151,12 @@ def as_text(ari:ari.ARI) -> str:
         enc.encode(ari, buf)
         return buf.getvalue()
 
+    def as_timespec(value:datetime) -> str:
+      seconds = value.total_seconds()
+      tv_sec = int(seconds)
+      tv_nsec = int(str(seconds).split(".")[1]) if "." in str(seconds) else 0 
+      return f"struct timespec ts; ts.tv_sec = {tv_sec}; ts.tv_nsec = {tv_nsec};"
```

Comment:
This can be made to use an inline struct field setter, which would avoid an implicit variable symbol name that could be a source of name collision depending on how it's used.